plugins/woocommerce/src/Admin/WCAdminHelper.php

```diff
@@ -122,4 +122,115 @@ public static function is_site_fresh() {
 
 		return $date > $month_ago;
 	}
+
+	/**
+	 * Test if a URL is a store page. This function ignores the domain and protocol of the URL and only checks the path and query string.
+	 *
+	 * Store pages are defined as:
+	 *
+	 * - My Account
+	 * - Shop
+	 * - Cart
+	 * - Checkout
+	 * - Privacy Policy
+	 * - Terms and Conditions
+	 *
+	 * Additionally, the following autogenerated pages should be included:
+	 * - Product pages
+	 * - Product Category pages
+	 * - Product Tag pages
+	 *
+	 * @param string $url URL to check. If not provided, the current URL will be used.
+	 * @return bool Whether or not the URL is a store page.
+	 */
+	public static function is_store_page( $url = '' ) {
+		$url = $url ? $url : esc_url_raw( wp_unslash( $_SERVER['REQUEST_URI'] ?? '' ) );
+
+		if ( ! $url ) {
+			return false;
+		}
+		$normalized_path = self::get_normalized_url_path( $url );
+
+		// WC store pages.
+		$store_pages = array(
+			'myaccount' => wc_get_page_id( 'myaccount' ),
+			'shop'      => wc_get_page_id( 'shop' ),
+			'cart'      => wc_get_page_id( 'cart' ),
+			'checkout'  => wc_get_page_id( 'checkout' ),
+			'privacy'   => wc_privacy_policy_page_id(),
+			'terms'     => wc_terms_and_conditions_page_id(),
+		);
+
+		foreach ( $store_pages as $page => $page_id ) {
+			if ( 0 >= $page_id ) {
+				continue;
+			}
+
+			$permalink = get_permalink( $page_id );
+			if ( ! $permalink ) {
+				continue;
+			}
+
+			if ( 0 === strpos( $normalized_path, self::get_normalized_url_path( $permalink ) ) ) {
+				return true;
+			}
+		}
+
+		// Check product, category and tag pages.
+		$permalink_structure = wc_get_permalink_structure();
```

Comment:
not sure how big of an impact it is on performance but there's a get_option() call in `wc_get_permalink_structure()`

Reply:
That's a good point. I think it's a minor impact. If it becomes a problem, we can optimize it later.

Reply:
From ChatGPT:

The impact of calling `wc_get_permalink_structure()` on performance depends on various factors such as server configuration, database size, and the context in which the function is being called.

1. **Caching**: If the result of `wc_get_permalink_structure()` is cached, subsequent calls to the function within the same request will have minimal impact on performance. WordPress itself caches the permalink structure, so in most cases, calling this function won't cause a significant performance hit.

2. **Database**: The function involves querying the database to retrieve the permalink structure. The impact may be noticeable on large or busy websites with heavy database loads. However, for most websites, especially those with caching mechanisms in place, the impact should be minimal.

3. **Context**: If the function is called frequently or in a critical performance path, such as within a loop that executes many times, the cumulative impact may become more significant. In such cases, it's advisable to call the function sparingly or optimize the code structure to minimize redundant calls.

4. **Server Resources**: The impact also depends on the server resources available. On high-traffic websites or under heavy server load, the impact may be more noticeable compared to low-traffic websites or under light server load.

Overall, calling `wc_get_permalink_structure()` typically has a minor impact on performance in most WordPress environments, especially when cached properly. However, it's essential to consider the specific context and optimize code accordingly if performance issues arise.